### ISSUE TITLE
Crash when capture_description of azurerm_eventhub removed

### ISSUE THREAD

**Issue by oxsteve** [hashicorp/terraform-provider-azurerm#15898]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

```
Terraform v1.1.7
on linux_amd64
+ provider registry.terraform.io/hashicorp/azurerm v2.99.0
```

### Affected Resource(s)

* `azurerm_eventhub`

### Terraform Configuration Files

```hcl
resource "azurerm_eventhub" "example" {
  name                = "acceptanceTestEventHub"
  namespace_name      = azurerm_eventhub_namespace.example.name
  resource_group_name = azurerm_resource_group.example.name
  partition_count     = 2
  message_retention   = 1

  # Crashes when this (optional) block is removed later on
  capture_description {
    enabled             = true
    encoding            = "Avro"

    destination {
      name                = "EventHubArchive.AzureBlockBlob"
      storage_account_id  = azurerm_storage_account.example.id
      blob_container_name = "example"
      archive_name_format = "{Namespace}/{EventHub}/{PartitionId}/{Year}/{Month}/{Day}/{Hour}/{Minute}/{Second}"
    }
  }
}
```

### Expected Behaviour

Capture is successfully removed from the Azure Event Hub. It should be in the same state as when the Capture feature has never been activated.

### Actual Behaviour

The plan correctly shows that the block will be removed. However, `terraform apply` crashes:

```
Stack trace from the terraform-provider-azurerm_v2.99.0_x5 plugin:

panic: runtime error: index out of range [0] with length 0

goroutine 130 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/eventhub.expandEventHubCaptureDescription(0x612dabd)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/eventhub/eventhub_resource.go:308 +0x77b
github.com/hashicorp/terraform-provider-azurerm/internal/services/eventhub.resourceEventHubUpdate(0x0, {0x5770660, 0xc00101fc00})
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/eventhub/eventhub_resource.go:232 +0x645
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc000f736b0, {0x67b1cf8, 0xc001b2ef00}, 0x24, {0x5770660, 0xc00101fc00})
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:363 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000b9a700, {0x67b1cf8, 0xc001b2ef00}, 0xc001817040, 0xc001b6f200, {0x5770660, 0xc00101fc00})
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:475 +0x6ba
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc001358d80, {0x67b1cf8, 0xc001b2ef00}, 0xc0004a5040)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:977 +0xd8a
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0001af280, {0x67b1da0, 0xc0019c5650}, 0xc00047a0e0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server/server.go:603 +0x30e
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x5cc9280, 0xc0001af280}, {0x67b1da0, 0xc0019c5650}, 0xc001802720, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001fefc0, {0x6802630, 0xc00146a000}, 0xc00184cfc0, 0xc0010ad9b0, 0xaf0b8a0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1292 +0xc6f
google.golang.org/grpc.(*Server).handleStream(0xc0001fefc0, {0x6802630, 0xc00146a000}, 0xc00184cfc0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1617 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:940 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:938 +0x294

Error: The terraform-provider-azurerm_v2.99.0_x5 plugin crashed!
```

### Steps to Reproduce

1. `terraform apply`
2. Remove the block `capture_description` from the `azurerm_eventhub`
3. `terraform apply`

### References

* #0000

**Comment by sinbai** [comment]
@oxsteve thank you for opening this issue. The PR [#15930](https://github.com/hashicorp/terraform-provider-azurerm/pull/15930) has been submitted to fix this bug.

**Comment by mbfrahry** [comment]
It looks like this was fixed in v3.0 of the provider. I'll close this down but feel free to reopen it if you're still facing issues.
